cmd/asset-syncer/utils.go

```diff
@@ -74,6 +75,27 @@ type assetManager interface {
 	Sync(charts []chart) error
 	RepoAlreadyProcessed(repoName, checksum string) bool
 	UpdateLastCheck(repoName, checksum string, now time.Time) error
+	Init() error
+	Close() error
+}
+
+func newManager(databaseType, databaseURL, databaseName, databaseUser, databasePassword string) (assetManager, error) {
+	if databaseType == "mongodb" {
+		mongoConfig := datastore.Config{URL: databaseURL, Database: databaseName, Username: databaseUser, Password: databasePassword}
+		return &mongodbAssetManager{mongoConfig, nil}, nil
+	} else if databaseType == "postgresql" {
```

Comment:
This could be a bit neater if each util (mongodb, postgresql) defined a `New` function, ie:
```
func NewMongoAssetManager(config *datastore.Config) (assetManager, error)
```
and similar for `NewPostgresAssetManager`. This would mean you don't need to instantiate objects with `nil` fields as you're doing here (more importantly, you don't need to know the implementation details, such as that they have a db field which needs to be nil on instantiation). It would also mean that it could initialize itself (returning the error).

So this whole section would be simply:
```
switch databaseType {
  case MONGODB:
    return NewMongoAssetManager(config)
  case POSTGROSQL:
    return NewPostgresAssetManager(config)
}
return nil, fmt.Errorf("Unsup....")
```